### ISSUE TITLE
when calling super method arguments are inherited from `object` instead of parent class

### ISSUE THREAD

**Issue by actionless** [python/mypy#7688]
Note: if you are reporting a wrong signature of a function or a class in
the standard library, then the typeshed tracker is better suited
for this report: https://github.com/python/typeshed/issues

Please provide more information to help us understand the issue:

* Are you reporting a bug, or opening a feature request?
Bug

* Please insert below the code you are checking with mypy,
  or a mock-up repro if the source is private. We would appreciate
  if you try to simplify your case to a minimal repro.

```python
import subprocess

class MyPopen(subprocess.Popen):

    def __init__(self) -> None:
        args = ['echo', 'test']
        super().__init__(
            args=args,
        )
```

* What is the actual behavior/output?
```
$ python -m mypy mypy_subprocess.py
mypy_subprocess.py:9: error: Unexpected keyword argument "args" for "__init__" of "object"
Found 1 error in 1 file (checked 1 source file)
```

* What are the versions of mypy and Python you are using?
```
$ python -m mypy --version
mypy 0.740+dev.742d33a41271206d6eb7dd2e4b04659de4683597

$ python --version
Python 3.7.4
```

**Comment by actionless** [comment]
NOTE: it was working fine with `mypy==0.720`

**Comment by ilevkivskyi** [comment]
This is a subtle problem in typeshed. @msullivan this is a one little additional argument in favour of supporting overloading on `self` in `__init__()` I proposed earlier, not that it is something we should do right now, but still.

I think it however makes sense to add a dummy `__init__()` to suppress this (if this will not cause other troubles). Also potentially we can move this issue to typeshed.

**Comment by ilevkivskyi** [comment]
I think this actually belongs in typeshed, I will try the new Github transfer issue feature now.
